notebooks/search/10-semantic-reranking.ipynb

```diff
@@ -0,0 +1,576 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b02700b4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    client.inference.delete_model(inference_id=\"cohere-rerank-inference\")\n",
+    "except exceptions.NotFoundError:\n",
+    "    # Inference endpoint does not exist\n",
+    "    pass\n",
+    "\n",
+    "try:\n",
+    "    client.options(\n",
+    "        request_timeout=60, max_retries=3, retry_on_timeout=True\n",
+    "    ).inference.put_model(\n",
+    "        task_type=\"rerank\",\n",
+    "        inference_id=\"cohere-rerank-inference\",\n",
+    "        model_config={\n",
+    "            \"service\": \"cohere\",\n",
+    "            \"service_settings\": {\n",
+    "                \"api_key\": \"kiOfsXJV2kMQ46xzve6snydXhKtfnfLI8pCgCdm0\",\n",
+    "                \"model_id\": \"rerank-english-v3.0\",\n",
+    "            },\n",
+    "        },\n",
+    "    )\n",
+    "    print(\"Inference endpoint created successfully\")\n",
+    "except exceptions.BadRequestError as e:\n",
+    "    if e.error == \"resource_already_exists_exception\":\n",
+    "        print(\"Inference endpoint created successfully\")\n",
+    "    else:\n",
+    "        raise e"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "818f7a72a83b5776",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "## Create the Index\n",
+    "\n",
+    "Now we need to create an index. Let's create one that enables us to perform search and semantic reranking on text articles."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ace87760606f67c6",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "client.indices.delete(index=\"semantic-reranking-articles\", ignore_unavailable=True)\n",
+    "client.indices.create(\n",
+    "    index=\"semantic-reranking-articles\",\n",
+    "    mappings={\n",
+    "        \"properties\": {\n",
+    "            \"title\": {\"type\": \"text\"},\n",
+    "            \"text\": {\"type\": \"text\"},\n",
+    "        },\n",
+    "    },\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2b5a46b60660a489",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "## Populate the Index\n",
+    "\n",
+    "Let's populate the index with a couple of random article fragments from Wikipedia."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "24f0133923553d28",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "url = \"https://raw.githubusercontent.com/elastic/elasticsearch-labs/main/notebooks/search/articles-wikipedia.json\"\n",
```

Comment:
If testing before this PR is merged, use this URL:
https://raw.githubusercontent.com/elastic/elasticsearch-labs/2b16684bce3abb31f0e6b20c9a52b0f3208baa2c/notebooks/search/articles-wikipedia.json